crates/red_knot_python_semantic/src/types.rs

```diff
@@ -3265,6 +3306,8 @@ pub(crate) mod tests {
         Union(Vec<Ty>),
         Intersection { pos: Vec<Ty>, neg: Vec<Ty> },
         Tuple(Vec<Ty>),
+        SubclassOfAny,
+        SubclassOfClass(&'static str),
```

Comment:
nit: looking at the implementation of the tests that use this variant, maybe it should be called `SubclassOfBuiltinClass`?